### PR TITLE
itest: export HashmailHarness struct

### PR DESCRIPTION
**Pull request by ViktorTigerstrom**
This PR exports the `HashmailHarness` struct, so that it can be used as an external dependency to setup a local mail server.

### CODE REVIEW

**Comment by ViktorTigerstrom** [comment]
> looks like we should maybe update the linter config. Linters like `exhaustivestruct` have been deprecated :)
Ok thanks, was a bit confused about what's happening with the CI! I will look into that @ellemouton!

**Comment by ViktorTigerstrom** [comment]
Ok finally got the linter to work properly :)! I updated the config to use a similar config to LND, as I figured it's likely our most up to date linter configs.

I pushed the update linter settings as 2 separate commits to this PR. Let me know if you think it's better that I push them as a separate PR instead @ellemouton!

File: itest/hashmailserver_harness.go [hunk 1]
```diff
@@ -20,6 +20,8 @@ type HashmailHarness struct {
 }
 
 // NewHashmailHarness creates a new instance of the HashmailHarness.
+//
+//nolint:exhaustruct
```
Comment (ellemouton) [inline itest/hashmailserver_harness.go:24]: i think we can just disable this linter

**Review by ellemouton — APPROVED** [review]
LGTM :) one nit :)

File: itest/hashmailserver_harness.go (hunk 1 above)
Comment (ViktorTigerstrom) [reply itest/hashmailserver_harness.go]: Ok thanks :)!

**Comment by ViktorTigerstrom** [comment]
Thanks for the review and the feedback @ellemouton! Addressed it with the latest push :)

**Review by bitromortac — APPROVED** [review]
LGTM :+1:, sorry for the late review

**Comment by ViktorTigerstrom** [comment]
Thanks for the review @bitromortac!

**Comment by ellemouton** [comment]
@ViktorTigerstrom  - the LNC repo prevents me from merging if the commits are not signed (see the "Merging is blocked" warning above)

**Comment by ViktorTigerstrom** [comment]
Updated the commits author to my GPG key email address & signed the commits.

**Pull request merged**